package/package.json

```diff
@@ -29,13 +29,25 @@
 	},
 	"type": "module",
 	"peerDependencies": {
-		"astro": "^4.0.0"
+		"@vitejs/plugin-react": "^4.2.1",
+		"astro": "^4.0.0",
+		"preact": "^10.19.4",
+		"react": "^18.2.0",
+		"solid-js": "^1.8.15",
+		"svelte": "^4.2.11",
+		"vue": "^3.4.19"
```

Comment:
I think those peer dependencies should be optional so a downstream lib is not required to install all of them.
Then the utilities could check if the needed dependencies are installed to give a nice error so it doesn't fail with the unexpected Vite/Node/Browser issues.

Reply:
@Fryuni  So I've just checked and we can't have optional peer dependencies, and having them as "optionalDependencies" will attempt to install them all whenever someone installs AIK. I think peer deps is the safer route here, unless there's something I'm missing? But I can definitely still add some checks in to make errors nicer

Reply:
There are peer dependencies, optional dependencies and optional peer dependencies.

This should be optional peer dependencies using the peer dependency meta.

Because ~~fucking JS ecosystem that's why~~:
- optional dependencies will always be installed but installation errors will be ignored, which is not ideal since it will install all of them
- peer dependencies are not optional by default, so dependents will be required to install them
- optional peer dependency will use the peer dependency, but won't me marked as missing dependency during installation

Reply:
https://docs.npmjs.com/cli/v10/configuring-npm/package-json#peerdependenciesmeta